plugin.video.vrt.nu/resources/lib/vrtplayer/urltostreamservice.py

```diff
@@ -75,33 +79,64 @@ def _get_playertoken(self, token_url, xvrttoken=None):
+            return xvrttoken
+        else:
+            return None
+        
+
     def _get_new_xvrttoken(self, path):
         cred = helperobjects.Credentials(self._kodi_wrapper)
         if not cred.are_filled_in():
             self._kodi_wrapper.open_settings()
             cred.reload()
-        data = {'loginID': cred.username, 'password': cred.password, 'APIKey': self._API_KEY, 'targetEnv': 'jssdk'}
+        data = {'loginID': cred.username, 'password': cred.password, 'sessionExpiration': '-1', 'APIKey': self._API_KEY, 'targetEnv': 'jssdk'}
         logon_json = requests.post(self._LOGIN_URL, data).json()
         if logon_json['errorCode'] == 0:
+            session = logon_json['sessionInfo']
             login_token = logon_json['sessionInfo']['login_token']
-            cookie_value = 'glt_'.join((self._API_KEY, '=', login_token))
-            payload = {'uid': logon_json['UID'], 'uidsig': logon_json['UIDSignature'], 'ts': logon_json['signatureTimestamp'], 'email': cred.password}
-            headers = {'Content-Type': 'application/json', 'Cookie': cookie_value}
+            login_cookie = ''.join(('glt_', self._API_KEY, '=', login_token))
+            payload = {'uid': logon_json['UID'], 'uidsig': logon_json['UIDSignature'], 'ts': logon_json['signatureTimestamp'], 'email': cred.username}
+            headers = {'Content-Type': 'application/json', 'Cookie': login_cookie}
             cookie_jar = requests.post(self._TOKEN_GATEWAY_URL, headers=headers, json=payload).cookies
             if 'X-VRT-Token' in cookie_jar:
                 xvrttoken_cookie = cookie_jar._cookies['.vrt.be']['/']['X-VRT-Token']
                 xvrttoken = { xvrttoken_cookie.name : xvrttoken_cookie.value, 'expirationDate' : datetime.datetime.fromtimestamp(xvrttoken_cookie.expires).strftime('%Y-%m-%dT%H:%M:%S.%fZ')}
-                json.dump(xvrttoken, open(path,'w'))
-                return xvrttoken['X-VRT-Token']
+                if 'roaming_XVRTToken' in path:
+                    xvrttoken = self._get_roaming_xvrttoken(login_cookie, xvrttoken)
+                    if xvrttoken is not None:
+                        json.dump(xvrttoken, open(path,'w'))
+                        return xvrttoken['X-VRT-Token']
+                    else:
+                        return None
+                else:
+                    json.dump(xvrttoken, open(path,'w'))
+                    return xvrttoken['X-VRT-Token']
             else:
                 return self._get_new_xvrttoken(path)
         else:
             title = self._kodi_wrapper.get_localized_string(32051)
             message = self._kodi_wrapper.get_localized_string(32052)
             self._kodi_wrapper.show_ok_dialog(title, message)
 
-    def _get_xvrttoken(self):
-        tokenfile = self._kodi_wrapper.get_userdata_path() + 'XVRTToken'
+    def _get_xvrttoken(self, tokenfile=None):
+        if tokenfile is None:
```

Comment:
rewrite 
        if tokenfile is None:
            tokenfile = self._kodi_wrapper.get_userdata_path() + 'XVRTToken'
        else:
            tokenfile = self._kodi_wrapper.get_userdata_path() + tokenfile

as tokenfile = self._kodi_wrapper.get_userdata_path() + (tokenfile or 'XVRTToken')